binding/react/src/use_porcupine.ts

```diff
@@ -9,209 +9,133 @@
     }
-    return false;
-  };
 
-  const stop = (): boolean => {
-    if (webVoiceProcessor !== null) {
-      webVoiceProcessor.stop().then(() => {
-        setIsListening(false);
-        return true;
-      });
+    try {
+      if (!porcupineRef.current) {
+        porcupineRef.current = await PorcupineWorker.create(
+          accessKey,
+          keywords,
+          keywordDetectionCallback,
+          model,
+          { ...options, processErrorCallback: errorCallback }
+        );
+        setIsLoaded(true);
+        setError(null);
+      }
+    } catch (e: any) {
+      setError(e.toString);
     }
-    return false;
-  };
+  }, [keywordDetectionCallback, errorCallback]);
 
-  const pause = (): boolean => {
-    if (webVoiceProcessor !== null) {
-      webVoiceProcessor.pause();
+  const start = useCallback(async (): Promise<void> => {
+    try {
+      if (!porcupineRef.current) {
+        setError("Porcupine not initialized");
+        return;
+      }
+
+      await WebVoiceProcessor.subscribe(porcupineRef.current);
+      setIsListening(true);
+      setError(null);
+    } catch (e: any) {
+      setError(e.toString());
       setIsListening(false);
-      return true;
     }
-    return false;
-  };
+  }, []);
 
-  const processErrorCallback = (error: string | Error): void => {
-    setIsError(true);
-    setErrorMessage(error.toString());
-  };
-
-  useEffect(() => {
-    // If using dynamic import() on porcupine-web-xx-worker,
-    // initially the worker factory may not exist yet; do nothing
-    if (
-      porcupineWorkerFactory === null ||
-      porcupineWorkerFactory === undefined
-    ) {
-      return (): void => {
-        /* NOOP */
-      };
-    }
+  const stop = useCallback(async (): Promise<void> => {
+    try {
+      if (!porcupineRef.current) {
+        setError("Porcupine not initialized");
+        return;
+      }
 
-    // When these are null, don't start Porcupine.
-    // If they previously were non-null, the useEffect cleanup will run and terminate the extant worker/webvp.
-    if (porcupineHookArgs === null || porcupineHookArgs === undefined) {
-      return (): void => {
-        /* NOOP */
-      };
+      await WebVoiceProcessor.unsubscribe(porcupineRef.current);
+      setIsListening(false);
+      setError(null);
+    } catch (e: any) {
+      setError(e.toString());
+      setIsListening(false);
     }
+  }, []);
 
-    const {
-      accessKey,
-      keywords,
-      start: startWebVp = true,
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    } = porcupineHookArgs!;
-
-    if (
-      keywords === null ||
-      keywords === undefined ||
-      (Array.isArray(keywords) && keywords.length === 0)
-    ) {
-      return (): void => {
-        /* NOOP */
-      };
-    }
+  const release = useCallback(async (): Promise<void> => {
+    if (porcupineRef.current) {
+      await stop();
+      porcupineRef.current.terminate();
```

Comment:
shouldn't we call `.release()` first? or does terminate handle that?

Reply:
terminate kills the worker so no need to call `.release()`